Directory.Build.props

```diff
@@ -1,12 +1,11 @@
 <Project>
   <Import Project="eng\Versions.props" />
   <PropertyGroup>
-    <BuildForWinUI Condition="'$(SolutionFileName)' == 'Microsoft.Maui.WinUI.sln'">true</BuildForWinUI>
-    <BuildForNet6 Condition="'$(SolutionFileName)' == 'Microsoft.Maui-net6.sln' or '$(BuildForWinUI)' == 'true'">true</BuildForNet6>
+    <BuildForNet6 Condition="'$(SolutionFileName)' == 'Microsoft.Maui-net6.sln'">true</BuildForNet6>
+    <BuildForWinUI Condition="'$(MSBuildRuntimeType)' == 'Full' And '$(SolutionFileName)' == 'Microsoft.Maui-net6.sln'">true</BuildForWinUI>
     <MauiPlatforms>net6.0-ios;net6.0-maccatalyst;net6.0-android</MauiPlatforms>
-    <WindowsTargetFramework Condition="'$(WindowsTargetFramework)' == ''">net6.0-windows10.0.18362</WindowsTargetFramework>
-    <MauiPlatforms Condition="'$(Packing)' == 'true'">$(MauiPlatforms);$(WindowsTargetFramework)</MauiPlatforms>
```

Comment:
If we get rid of `$(Packing)` (which is good), I think we can remove it here, too:

https://github.com/dotnet/maui/blob/ad8980a1fc9e892ddf980b3da731002c5100dce1/eng/package.ps1#L59